src/Components/MediaPresenter.ts

```diff
@@ -1,42 +1,58 @@
+import * as mobx from 'mobx';
+
 const MAXLEN = 10;
 
+
 export class MediaPresenter {
+  constructor() {
+    mobx.makeObservable(this);
+  }
+  // make need to change to .deep
+  @mobx.observable.deep
   files: File[] = [];
+
+  @mobx.observable.deep
   durations: number[] = [];
-  customOrder: boolean = false;
 
-  static imageFormat = new RegExp("image/*");
-  static videoFormat = new RegExp("video/*");
+  customOrder: boolean = false;
 
   // this is temporary, will remove
   imageDuration: number = 3000;
   static audioSound: number = 0.5;
 
-  addFile(newFile: File): void {
-    if (MediaPresenter.isImage(newFile) || MediaPresenter.isVideo(newFile)) {
-      this.files = [...this.files, newFile];
-    } 
-  }
-
   static isImage(file: File) {
-    return MediaPresenter.imageFormat.test(file.type);
+    const imageFormat = new RegExp("image/*");
+    console.log("isimage", file);
+    return imageFormat.test(file.type);
   }
 
   static isVideo(file: File) {
-    return MediaPresenter.videoFormat.test(file.type);
+    const videoFormat = new RegExp("video/*");
+    console.log("isvideo", file);
+    return videoFormat.test(file.type);
   }
+
+  @mobx.action
+  addFile(newFile: File): void {
```

Comment:
> Quiz question: why can't this be a `ref`?

because the way I have implemented addFile here is not changing the reference but adding values into the array (?) Therefore .deep is required (or just @observable)

So if I have implemented like so, then it would work with ref.
  addFile(newFile: File): void {
    if (MediaPresenter.isImage(newFile) || MediaPresenter.isVideo(newFile)) {
      this.files = [...this.files, newFile];
    } 
  }

Reply:
@juliamitchelmore thoughts on this? :)